### DISCUSSION TITLE
Prisma findMany where clause behavior with undefined values

### DISCUSSION THREAD

**Discussion by mortezaisvand** [prisma/prisma#21384] [Q&A]
### Question

Hello everyone,

I’m currently working on a project where I’m using Prisma to interact with my database. I have a situation where I want to fetch posts based on a search term. Here’s the code snippet:
```typescript
import prisma from "@/prisma/client";
import { NextRequest, NextResponse } from "next/server";

export async function GET(req: NextRequest) {
  const { searchParams } = new URL(req.url);
  const search = searchParams.get("search") || "";
  try {
    const posts = await prisma.post.findMany({
      where: {
        title:
          search != undefined
            ? {
                contains: search,
                mode: "insensitive",
              }
            : undefined,
      },
      select: {
        id: true,
        title: true,
        content: true,
        tag: true,
      },
    });
    return NextResponse.json(posts, { status: 200 });
  } catch (error) {
    return NextResponse.json(
      { message: "Could not fetch posts" },
      { status: 500 }
    );
  }
}


```

In the `where` clause of the findMany function, I want to include the `where` condition only when the `search` parameter is not `undefined`. When `search` is undefined I want the `where` clause to have no effect at all. 

The above code doesn't work as expected. As a matter of fact, when the `search` is undefined nothing will be rendered in the home page, but when it is a string or an empty string all posts will be fetched.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hello @MortezaIsvand 👋 

Did you had a look at this section of docs: [Undefined and Null](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined)

The behavior you're experiencing is due to how Prisma handles `undefined` values in conditionals. When you pass undefined to a filter, Prisma treats it as if no filter was applied.

Can you do something like this and check if you get the correct behaviour?

```ts
import prisma from "@/prisma/client";
import { NextRequest, NextResponse } from "next/server";

export async function GET(req: NextRequest) {
  const { searchParams } = new URL(req.url);
  const search = searchParams.get("search") || "";
  let whereClause = {};
  if (search) {
    whereClause = {
      title: {
        contains: search,
        mode: "insensitive",
      },
    };
  }
  try {
    const posts = await prisma.post.findMany({
      where: whereClause,
      select: {
        id: true,
        title: true,
        content: true,
        tag: true,
      },
    });
    return NextResponse.json(posts, { status: 200 });
  } catch (error) {
    return NextResponse.json(
      { message: "Could not fetch posts" },
      { status: 500 }
    );
  }
}
```

**Reply by mortezaisvand** [reply]
I tried your code, but nothing happened. I have already provided full details of my code in the discussion, but nobody has responded.

**Comment by mortezaisvand** [comment]
According to the Prisma docs, if `undefined` means `do nothing`, therefore the `where` clause should be ignored and the code should function as expected.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
